sockeye/arguments.py

```diff
@@ -1139,6 +1139,14 @@ def add_inference_args(params):
                                type=int,
                                default=None,
                                help='Maximum input sequence length. Default: value from model(s).')
+    decode_params.add_argument('--use-dynamic-batch-size',
```

Comment:
How about instead of having a separate parameter enabling a dynamic batch size by default. Namely, so that `Translator.translate` would internally use a reduced batch_size for each `List[TranslatorInput]` that is smaller than the `batch_size`.  Then the user wouldn't need to specify any option and the `Translator` object would automatically do what's most efficient.

For input coming from stdin it is less clear why one would want dynamic batching, as in the current stage it is mainly aimed at batch translation where large chunks are read, sorted and translated. Then again, if we changed the Translator as described above, the last batch in a chunk (which might be smaller than `batch_size` could make use of dynamic batching. If we consider the realm time translation use case from stdin, it might make sense to define a json list of inputs interface, for which then in turn could use dyanmic batching.